### ISSUE TITLE
[Request] Ardupilot OSD icon support

### ISSUE THREAD

**Issue by lida2003** [OpenIPC/msposd#13]
It's different from OSD tab, any idea? Pitch/Roll value  is wrong, should be 1 ~5 degree, as it's leveled.

- APMA 4.5.6
- OpenIPC master+8d995d7, 2024-10-19

![图片](https://github.com/user-attachments/assets/5a9a7786-f85c-45b5-91ae-a224bda3d400)

![图片](https://github.com/user-attachments/assets/9d4ffa77-9f65-45c0-ada8-b1aa39c092a8)

**Comment by lida2003** [comment]
OK, I figured out. msposd do NOT have ardupilot png files.

- https://sneaky-fpv.gitbook.io/hd-osd-and-tools

![图片](https://github.com/user-attachments/assets/6fa74ae4-8100-4920-8039-1a220d126638)

**Comment by tipoman9** [comment]
Most users reported they can use Betaflight fonts when setting ```MSP_OPTIONS fonts = 4``` in ardupilot.
Can you summarize what config changes are needed to make ArduPilot use its native font?

**Comment by lida2003** [comment]
**Analog Camera:**

```
SERIAL3_BAUD = 230     //230400
SERIAL3_PROTOCOL = 44  //IRC_TRAMP
SERIAL3_OPTIONS = 4    //halfDuplex
OSD_TYPE1 = 1          //MAX7456
```

**Digital Camera:**

```
SERIAL8_BAUD = 115     //115200
SERIAL8_PROTOCOL = 42  //DISPLAYPORT
SERIAL8_OPTIONS = 0    //polling
OSD_TYPE2 = 5          //MSP_DISPLAYPORT
MSP_OPTIONS = 4        //EnableBTFLFonts
```

Above settings for two camera works, and I think a more convient local postion feature is much more handy in my case. But I think there are not that many people use. Anyway, I have post a request here. 


- [ OSD icon local position configuration support #14 ](https://github.com/OpenIPC/msposd/issues/14)

**Comment by tipoman9** [comment]
So the above options work with Betaflight fonts, right?
What settings are needed to make ardupilot use its native fonts ?

**Comment by lida2003** [comment]
> What settings are needed to make ardupilot use its native fonts ?

I can't make it work, even the initial icons are displayed as garbled image. And there is a bug I think https://github.com/OpenIPC/msposd/issues/15

**Comment by tipoman9** [comment]
Well, several users reported ardupilot latest version working with Betaflight fonts with these settings:
```OSD_TYPE=5
SERIALx_PROTOCOL=42
MSP_OPTIONS=6```

**Comment by lida2003** [comment]
I don't have any idea about MSP_OPTIONS=6. Did you get feedback from developer or users, as I couldn't find any links about OpenIPC configuration, especially those unused settings.

![图片](https://github.com/user-attachments/assets/7c4b2ac8-f367-45de-8f52-14be18dc27a1)

**Comment by tipoman9** [comment]
As I explained you here, https://github.com/OpenIPC/msposd/issues/15
Ardupilot can be configured to use Betaflight fonts.
If you have trouble with this, please ask in Ardupilot channel for help.

**Comment by lida2003** [comment]
> Ardupilot can be configured to use Betaflight fonts.

No, msposd doesn't support native ardupilot display port. As the incon index is different. Then you use a betaflight configuration option, but the power icon is wrong . Then you said it's png icon index is different fron betaflight. 

So I'm NOT sure if it's png issue or msposd code issue or ardupilot's code issue on msp option. Anyway, it didn't work with that icon, am i right?

**Comment by tipoman9** [comment]
_Currently msposd is not tested with ardupilots "native" fonts, since the demand for this was lacking.
Several users reported that they can configure ardupilot to output osd icon index for Betaflight fonts.
I hope I'll be able to add support for native Ardu fonts when I find a spare flight controller to install it and test it. (apart from the two I use to test Inav and Betaflight)_

There is no "VTX power icon" in Betaflight fonts, so when you configure Ardupilot to use Betaflight fonts it simply takes the glyph from the location it is supposed to be in Ardu fonts I suppose.
Since this icon is so important for you, you can manually copy it in the betaflight font png.
